### PR TITLE
lint: Enable global strict eslint rule

### PR DESCRIPTION
**Pull request by scagood**
References:
https://github.com/eslint-community/eslint-plugin-n/pull/180#discussion_r1502190641
#190

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -37,4 +39,11 @@ module.exports = [
             "eslint-plugin/no-unused-message-ids": 0,
         },
     },
+    {
+        // these messageIds were used outside
+        files: ["lib/**/*.js"],
```
Comment (aladdin-add) [inline eslint.config.js:44]:
```suggestion
```
this should be applied to all js files?

Comment (aladdin-add) [inline eslint.config.js:44]:
```suggestion
```
this should be applied to all js files?

Comment (scagood) [reply eslint.config.js:44]: I didnt want to mess with the test fixtures :see_no_evil: you're right though!

Comment (aladdin-add) [reply eslint.config.js:44]: ignore: ["fixtures/"]?

**Pull request merged**